vllm/worker/utils.py

```diff
@@ -47,10 +47,6 @@ def assert_enc_dec_mr_supported_scenario(
         raise NotImplementedError(
             STR_NOT_IMPL_ENC_DEC_ERR_STRS['STR_NOT_IMPL_ENC_DEC_SPEC_DEC'])
 
-    if not enc_dec_mr.model_config.enforce_eager:
-        raise NotImplementedError(
-            STR_NOT_IMPL_ENC_DEC_ERR_STRS['STR_NOT_IMPL_ENC_DEC_CUDA_GRAPH'])
-
```

Comment:
Overall this PR looks great! I went through the codebase to find any additional mention of CUDAGraph not being supported for encoder/decoder models. I find some occurrences which should probably be addressed by this PR.

1. In `vllm/utils.py` I recommend also removing `STR_NOT_IMPL_ENC_DEC_CUDAGRAPH` as well as any logic which depends on it:

https://github.com/sroy745/vllm/blob/7b94a359692bde3de9e00eb76ce985a92fae71ed/vllm/utils.py#L71

2. In the LLM entrypoint code there is a comment which we probably want to remove or change:

```
        LLM constructor.

        Note: if enforce_eager is unset (enforce_eager is None)
        it defaults to False for decoder-only models and True
        for encoder/decoder models, since encoder/decoder models
        do not currently support CUDAGraph.
```

https://github.com/sroy745/vllm/blob/7b94a359692bde3de9e00eb76ce985a92fae71ed/vllm/entrypoints/llm.py#L136-L143

3. Same here in `ModelConfig`

```
        enforce_eager: Whether to enforce eager execution. If True, we will
            disable CUDA graph and always execute the model in eager mode.
            If False, we will use CUDA graph and eager execution in hybrid.
            If None, the user did not specify, so default to False -
            except for encoder/decoder models, which currently require
            eager mode.
```

https://github.com/sroy745/vllm/blob/7b94a359692bde3de9e00eb76ce985a92fae71ed/vllm/config.py#L94-L96